apps/io-wallet-user-func/src/android-key-attestation.ts

```diff
@@ -0,0 +1,127 @@
+import * as appInsights from "applicationinsights";
+import { X509Certificate } from "crypto";
+import * as E from "fp-ts/lib/Either";
+import * as RTE from "fp-ts/lib/ReaderTaskEither";
+import * as TE from "fp-ts/lib/TaskEither";
+import { pipe } from "fp-ts/lib/function";
+import { WalletInstanceValid } from "io-wallet-common/wallet-instance";
+
+import { AttestationServiceConfiguration } from "./app/config";
+import {
+  validateIssuance,
+  validateRevocation,
+} from "./infra/attestation-service/android/attestation";
+import {
+  WalletInstanceRepository,
+  getAllValidWalletInstances,
+  revokeUserWalletInstances,
+} from "./wallet-instance";
+
+type WalletInstanceValidWithAndroidCertificatesChain = {
+  deviceDetails: { platform: string; x509Chain: string[] };
+} & WalletInstanceValid;
+
+const hasAndroidCertificatesChain = (
+  walletInstance: WalletInstanceValid,
+): walletInstance is WalletInstanceValidWithAndroidCertificatesChain =>
+  !!walletInstance.deviceDetails &&
+  walletInstance.deviceDetails.platform === "android" &&
+  !!walletInstance.deviceDetails.x509Chain;
```

Comment:
```suggestion
  walletInstance.deviceDetails?.platform === "android" &&
  walletInstance.deviceDetails.x509Chain;
```

Reply:
typescript fa i capricci: 
> **Type 'false | string[] | undefined' is not assignable to type 'boolean'.
  Type 'undefined' is not assignable to type 'boolean'.**